backend/src/api/courses/courses.api.ts

```diff
@@ -17,8 +17,19 @@ const initCoursesApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
   fastify.route({
     method: HttpMethod.GET,
     url: CoursesApiPath.ROOT,
-    async handler(_req, rep) {
-      const courses = await courseService.getAll();
+    async handler(
+      req: FastifyRequest<{ Querystring: { title: string } }>,
```

Comment:
And provide validation for filtering opts in query
```suggestion
      req: FastifyRequest<{ Querystring: CourseFilteringDto }>,
```